### PR TITLE
Increase HTTP server backlog size to 2

### PR DESCRIPTION
**Pull request by jimmyatwork**
## Description

This PR increases the backlog size of the ad hoc HTTP server from `1` to `2`.

## Motivation and Context

I'm using `BrowserSamlCredentialsProvider` in [DataGrip](https://www.jetbrains.com/datagrip/) to connect to Redshift. I noticed that the login process would randomly fail in **Google Chrome on macOS** with `ERR_SOCKET_NOT_CONNECTED`:

Screen Shot 2023-08-21 at 11 46 38 AM

The error appears to be related to Chrome's "Preload pages" feature,
located at `chrome://settings/preloading`:

Screen Shot 2023-08-21 at 12 38 37 PM

It causes Chrome to make an extra socket connection that frequently (but
not always) occupies the backlog size of `1` and causes the
subsequent `/redshift/` HTTP request to fail with
`ERR_SOCKET_NOT_CONNECTED`.

The error does not happen if preloading is disabled.

Since preloading is enabled by default, I thought it would be better to
increase the backlog size than to ask the user to disable preloading.

Increasing the backlog size to `2` seems to be enough to accommodate the
extra "preload" socket connection and allow the `/redshift/`
request to succeed.

## Testing

- I'm using macOS 12.6.6, running on Apple M1 Max
- Chrome is my default browser

I reproduced the error and verified the fix by running this code multiple times:

```java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

class App {
    public static void main(String []args) throws SQLException, ClassNotFoundException {
        Class.forName("com.amazon.redshift.jdbc.Driver");
        Properties props = new Properties();
        props.setProperty("plugin_name", "com.amazon.redshift.plugin.BrowserSamlCredentialsProvider");
        props.setProperty("login_url", "<LOGIN_URL>");
        DriverManager.getConnection("jdbc:redshift:iam://<CLUSTER_NAME>:us-west-2/dev", props);
    }
}
```

I also tested a local build of the driver in DataGrip.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist

- [x] Local run of `mvn install` succeeds
- [x] My code follows the code style of this project
- [ ] My change requires a change to the Javadoc documentation
- [ ] I have updated the Javadoc documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

- [x] By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by bhvkshah** [comment]
Thanks for the PR and the detailed summary, @jimmyatwork ! The changes look good, I'll give it another round of reviews internally before merging it in.

File: src/main/java/com/amazon/redshift/plugin/httpserver/Server.java [hunk 1]
```diff
@@ -90,7 +90,10 @@ public Server(int port, RequestHandler handler, Duration waitTime,
         this.m_ipAddress = InetAddress.getLoopbackAddress();
         this.m_socketFactory = ServerSocketFactory.getDefault();
         this.m_defaultSocketConfig = SocketConfig.custom()
-            .setBacklogSize(1)
+            // The backlog size needs to be at least 2 to avoid random `ERR_SOCKET_NOT_CONNECTED`
+            // errors in Google Chrome on macOS. It's caused by Chrome's "Preload pages" feature,
+            // which is enabled by default (located here: `chrome://settings/preloading`).
+            .setBacklogSize(2)
```
Comment (bhvkshah) [inline src/main/java/com/amazon/redshift/plugin/httpserver/Server.java:96]: @jimmyatwork are there any performance impacts to setting backlog size to 2 here?

**Review by bhvkshah — APPROVED** [review]
Looks good. These changes will be officially available in driver version 2.1.0.19. Thank you for your contribution to the Redshift JDBC Driver @jimmyatwork !

**Comment by jimmyatwork** [comment]
Thank you @bhvkshah, I appreciate the quick review!

**Pull request merged**